src/ashell/jerry-code.c

```diff
@@ -74,6 +120,100 @@ static void javascript_print_value(const jerry_value_t value)
     jerry_port_console("\n");
 }
 
+static int read_file(const char *file_name, char **file_buf)
+{
+    fs_file_t *fp = fs_open_alloc(file_name, "r");
+
+    if (fp == NULL)
+        return 0;
+
+    ssize_t size = fs_size(fp);
+    if (size == 0) {
+        comms_printf("[ERR] Empty file (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    *file_buf = (char *)zjs_malloc(size);
+    if (*file_buf == NULL) {
+        comms_printf("[ERR] Not enough memory for (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    ssize_t brw = fs_read(fp, *file_buf, size);
+
+    if (brw != size) {
+        comms_printf("[ERR] Failed loading code %s\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    return size;
+}
+
+static void find_requires(char *filebuf)
+{
+    char *ptr1 = filebuf;
+    char *ptr2 = NULL;
+    char *filestr = NULL;
+    char *ext_check = NULL;
+    while(ptr1 != NULL){
+
+        // Find next instance of "require"
+        ptr1 = strstr(ptr1, "require");
+
+        if (ptr1 != NULL) {
+            // Find the text between the two " " after require
+            ptr1 = strchr(ptr1, '"')+1;
+            ptr2 = strchr(ptr1, '"');
+            size_t len = ptr2-ptr1;
+
+            // Allocate the memory for the string
+            filestr = (char*)zjs_malloc(len+1);
+            strncpy(filestr, ptr1, len);
+            filestr[len] = '\0';
+
+            // Check that this is a *.js require and not a zephyr require
+            // Also check that the file hasn't already been loaded
+            ext_check = &filestr[len-3];
+
+            if (strcmp(ext_check, ".js") == 0 && !is_loaded(filestr))
+            {
```

Comment:
Move brace to previous line.

Personally, I use !strcmp() for "equal" or just strcmp() for "not equal" rather than comparing with zero. I find that easier to read; when I see this I have to sit and think about it. But maybe that's just habit.